astropy/units/quantity.py

```diff
@@ -738,7 +763,7 @@ def to_value(self, unit=None, equivalencies=[]):
         """
         if unit is None or unit is self.unit:
             value = self.view(np.ndarray)
-        else:
+        elif not self.dtype.names:
```

Comment:
```suggestion
        elif not self.dtype.names:   # not structured
```

Reply:
Adapted with minor change (# for non-structured, we attempt a short-cut.)